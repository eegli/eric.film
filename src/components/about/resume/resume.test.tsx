import { fireEvent, render } from '@/src/tests/testUtils';
import Resume from './resume.component';

describe('About', () => {
  it('renders resume component', () => {
    const { asFragment } = render(<Resume />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('opens education, work on click', () => {
    const { asFragment, getByText } = render(<Resume />);
    fireEvent.click(getByText('education - work'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('opens tools, technology on click', () => {
    const { asFragment, getByText } = render(<Resume />);
    fireEvent.click(getByText('tools - technology'));
    expect(asFragment()).toMatchSnapshot();
  });
});
