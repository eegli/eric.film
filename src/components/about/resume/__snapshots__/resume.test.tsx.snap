// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About opens education, work on click 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 0.5rem;
  width: 93%;
  color: #121212;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 1rem;
}

.c1:nth-child(2) {
  margin-left: 1rem;
  margin-right: 0;
}

.c4 {
  background-color: #222222;
  color: #f7f7f7;
  padding: 1rem 2rem 1rem 1rem;
}

.c4 span {
  color: #F1F1F1;
  font-size: 14px;
}

.c4 > div {
  margin-top: 0.5rem;
  padding: 0.6rem;
}

.c4 > div > p {
  margin: 0 0 0.8rem 0;
}

.c4 > div > ul {
  margin: 0 0 0.8rem 0;
  padding-left: 1.2rem;
}

.c4 > div * li {
  margin-bottom: 0.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #3b3b3b;
  color: #F1F1F1;
  padding: 1rem;
}

.c2:hover {
  cursor: pointer;
}

.c3 {
  display: inline;
  margin: 0;
  font-weight: 500;
}

@media screen and (max-width:1000px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (max-width:1000px) {
  .c1:nth-child(2) {
    margin: 2rem 0 0 0;
  }
}

@media screen and (max-width:1000px) {
  .c1 {
    margin: 0;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          education - work
        </p>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
          />
        </svg>
      </div>
      <div
        class=""
        style="height: 0px; overflow: hidden; transition: height .25s ease;"
      >
        <div
          style="overflow: hidden;"
        >
          <div
            class="c4"
          >
            <span>
              EDUCATION
            </span>
            <div>
              <span>
                2020 - 2024
              </span>
              <p>
                Master of Science, Data Science. University of Zurich
              </p>
              <span>
                2015 - 2019
              </span>
              <p>
                Bachelor of Arts, Media and Communication Science. University of Zurich
              </p>
              <span>
                since 2018
              </span>
              <p>
                Various courses on Udemy: Javascript, Typescript, React.js, React native, HTML, CSS
              </p>
            </div>
            <span>
              WORK
            </span>
            <div>
              <span>
                05.2018 - present
              </span>
              <p>
                Digital Project Manager. Ringier AG
              </p>
              <span>
                01.2013 - present
              </span>
              <p>
                Freelance photographer & videographer. Self-employed
              </p>
              <span>
                01.2017 - 04.2018
              </span>
              <p>
                Project Management Support. Ringier AG
              </p>
              <span>
                05.2016 - 04.2019
              </span>
              <p>
                Video Journalist. Zürcher Oberländer Medien AG
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          tools - technology
        </p>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          />
        </svg>
      </div>
      <div
        class=""
        style="height: 0px; overflow: hidden; display: none; transition: height .25s ease;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`About opens tools, technology on click 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 0.5rem;
  width: 93%;
  color: #121212;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 1rem;
}

.c1:nth-child(2) {
  margin-left: 1rem;
  margin-right: 0;
}

.c4 {
  background-color: #222222;
  color: #f7f7f7;
  padding: 1rem 2rem 1rem 1rem;
}

.c4 span {
  color: #F1F1F1;
  font-size: 14px;
}

.c4 > div {
  margin-top: 0.5rem;
  padding: 0.6rem;
}

.c4 > div > p {
  margin: 0 0 0.8rem 0;
}

.c4 > div > ul {
  margin: 0 0 0.8rem 0;
  padding-left: 1.2rem;
}

.c4 > div * li {
  margin-bottom: 0.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #3b3b3b;
  color: #F1F1F1;
  padding: 1rem;
}

.c2:hover {
  cursor: pointer;
}

.c3 {
  display: inline;
  margin: 0;
  font-weight: 500;
}

@media screen and (max-width:1000px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (max-width:1000px) {
  .c1:nth-child(2) {
    margin: 2rem 0 0 0;
  }
}

@media screen and (max-width:1000px) {
  .c1 {
    margin: 0;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          education - work
        </p>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          />
        </svg>
      </div>
      <div
        class=""
        style="height: 0px; overflow: hidden; display: none; transition: height .25s ease;"
      />
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          tools - technology
        </p>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
          />
        </svg>
      </div>
      <div
        class=""
        style="height: 0px; overflow: hidden; transition: height .25s ease;"
      >
        <div
          style="overflow: hidden;"
        >
          <div
            class="c4"
          >
            <span>
              DEVELOPMENT
            </span>
            <div>
              <ul>
                <li>
                  Javascript (ES5, ES6)
                </li>
                <li>
                  Typescript
                </li>
                <li>
                  Next.js
                </li>
                <li>
                  React.js, React native
                </li>
                <li>
                  Redux, React Redux
                </li>
                <li>
                  CSS in JS
                </li>
                <li>
                  HTML5, CSS3
                </li>
                <li>
                  GraphQL, REST
                </li>
                <li>
                  Google Firebase
                </li>
                <li>
                  Git
                </li>
                <li>
                  Node.js (currently learning)
                </li>
              </ul>
            </div>
            <span>
              MISC
            </span>
            <div>
              <ul>
                <li>
                  Agile (Scrum & Kanban)
                </li>
                <li>
                  Adobe Creative Suite
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`About renders resume component 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 0.5rem;
  width: 93%;
  color: #121212;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-right: 1rem;
}

.c1:nth-child(2) {
  margin-left: 1rem;
  margin-right: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #3b3b3b;
  color: #F1F1F1;
  padding: 1rem;
}

.c2:hover {
  cursor: pointer;
}

.c3 {
  display: inline;
  margin: 0;
  font-weight: 500;
}

@media screen and (max-width:1000px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (max-width:1000px) {
  .c1:nth-child(2) {
    margin: 2rem 0 0 0;
  }
}

@media screen and (max-width:1000px) {
  .c1 {
    margin: 0;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          education - work
        </p>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          />
        </svg>
      </div>
      <div
        class=""
        style="height: 0px; overflow: hidden; display: none; transition: height .25s ease;"
      />
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          tools - technology
        </p>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          />
        </svg>
      </div>
      <div
        class=""
        style="height: 0px; overflow: hidden; display: none; transition: height .25s ease;"
      />
    </div>
  </div>
</DocumentFragment>
`;
